pkg/service/did/key.go

```diff
@@ -99,7 +100,26 @@ func (h *keyDIDHandler) GetDIDs(ctx context.Context, method did.Method) (*GetDID
 	}
 	dids := make([]did.DIDDocument, 0, len(gotDIDs))
 	for _, gotDID := range gotDIDs {
-		dids = append(dids, gotDID.DID)
+		if !gotDID.SoftDeleted {
+			dids = append(dids, gotDID.DID)
+		}
 	}
 	return &GetDIDsResponse{DIDs: dids}, nil
 }
+
+func (h *keyDIDHandler) SoftDeleteDID(ctx context.Context, request DeleteDIDRequest) error {
+	logrus.Debugf("soft deleting DID: %+v", request)
+
+	id := request.ID
+	gotStoredDID, err := h.storage.GetDID(ctx, id)
+	if err != nil {
+		return fmt.Errorf("error getting DID: %s", id)
+	}
+	if gotStoredDID == nil {
+		return fmt.Errorf("did with id<%s> could not be found", id)
+	}
+
+	gotStoredDID.SoftDeleted = true
```

Comment:
Do we need to do optimistic locking for this update? Is it possible to modify an already stored did within out system?

Reply:
I dont think we need locking here since the flipping of SoftDeleted doesn't rely on any other entity.
Currently we can only create, get and delete dids, so no updating can happen.

Reply:
Will we want to support updating DIDs? For example, what if someone wants to do an ION operation?

Deletion then would compete against an update operation, and we could end up losing some data. That sounds like something we'll be adding soon(ish). So I would safe better to be safe than sorry.

Reply:
If the ion did get updated in the bitcoin 'database' and ours only points to it I think it will still be fine I think, its only if we do any updates to existing dids in our DB.

I made an issue regardless and yea it couldn't hurt to throw it in there. We should have a bigger store to use our optimistic locking in many places. - https://github.com/TBD54566975/ssi-service/issues/336